pinot-spi/src/main/java/org/apache/pinot/spi/env/CommonsConfigurationUtils.java

```diff
@@ -100,24 +107,50 @@ public static PropertiesConfiguration fromPath(String path, boolean setIOFactory
    * @return a {@link PropertiesConfiguration} instance.
    */
   public static PropertiesConfiguration fromInputStream(InputStream stream, boolean setIOFactory,
-      boolean setDefaultDelimiter)
+      boolean setDefaultDelimiter, PropertyIOFactoryKind ioFactoryKind)
       throws ConfigurationException {
-    PropertiesConfiguration config = createPropertiesConfiguration(setIOFactory, setDefaultDelimiter);
+    PropertiesConfiguration config = createPropertiesConfiguration(setIOFactory, setDefaultDelimiter, ioFactoryKind);
     FileHandler fileHandler = new FileHandler(config);
     fileHandler.load(stream);
     return config;
   }
 
+  /**
+   * Instantiate a Segment Metadata {@link PropertiesConfiguration} from a {@link File}.
+   * @param file containing properties
+   * @param setIOFactory representing to set the IOFactory or not
+   * @param setDefaultDelimiter representing to set the default list delimiter.
+   * @return a {@link PropertiesConfiguration} instance.
+   */
+  public static PropertiesConfiguration segmentMetadataFromFile(File file, boolean setIOFactory,
+      boolean setDefaultDelimiter)
+      throws ConfigurationException {
+    PropertyIOFactoryKind ioFactoryKind = PropertyIOFactoryKind.DefaultPropertyConfigurationIOFactory;
+
+    try {
+      if (checkHeaderExistsInSegmentMetadata(file)) {
+        ioFactoryKind = PropertyIOFactoryKind.SegmentMetadataIOFactory;
+      }
+    } catch (IOException exception) {
+      throw new ConfigurationException(
+          String.format("Error occurred while reading segment metadata file %s ", file.getName()), exception);
+    }
+
+    return fromFile(file, setIOFactory, setDefaultDelimiter, ioFactoryKind);
+  }
+
   /**
    * Instantiate a {@link PropertiesConfiguration} from a {@link File}.
    * @param file containing properties
    * @param setIOFactory representing to set the IOFactory or not
    * @param setDefaultDelimiter representing to set the default list delimiter.
    * @return a {@link PropertiesConfiguration} instance.
    */
-  public static PropertiesConfiguration fromFile(File file, boolean setIOFactory, boolean setDefaultDelimiter)
+  public static PropertiesConfiguration fromFile(File file, boolean setIOFactory,
+      boolean setDefaultDelimiter, PropertyIOFactoryKind ioFactoryKind)
       throws ConfigurationException {
-    PropertiesConfiguration config = createPropertiesConfiguration(setIOFactory, setDefaultDelimiter);
+    Preconditions.checkNotNull(file, "File object can not be null for loading configurations");
```

Comment:
do we need to do null check? I don't feel so, but if there is need, let's annotate `@Nullable File file`.